sigstore/sign.py

```diff
@@ -387,7 +411,8 @@ def to_bundle(self) -> Bundle:
             verification_material=material,
             message_signature=MessageSignature(
                 message_digest=HashOutput(
-                    algorithm=HashAlgorithm.SHA2_256,
+                    #algorithm=HashAlgorithm.SHA2_256,
+                    algorithm=self.digest_algorithm,
```

Comment:
Can you talk me through this (possibly I just don't understand the protobuf details):
* we're creating protobuf message HashOutput where algorithm is a string (that comes from cryptography?)
* but the protobuf HashAlgorithm is actually an enum 
* HashAlgorithm has a single documented value (SHA2_256)

Reply:
HashAlgorithm comes from Sigstore `sigstore_protobuf_specs.dev.sigstore.common.v1` which comes from `https://github.com/sigstore/protobuf-specs/blob/d96fa4407cb4489233f133ab036c0ed98c2756a4/protos/sigstore_common.proto#L39`. The value is a string, ifaict.

Reply:
I don't think it works that way. The python bindings might not complain when that r.uns it but the protobuf serialization likely won't work.

But I suppose if you were asking for high level feedback, the real questions related to this are: 
* how is this going to work in bundle that is an interop format? 
* should the bundle even try to document the used algorithm (since the verifier I guess is anyway expected to know the magic precalculation method that is required to actually verify) -- does the protobuf need a "custom" algorithm value?

Reply:
re: yeah you're right about types I double checked. I replied to your question about bundle in the other comment. TL;DR: We're adding a hash in https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_common.proto#L39 which will rekor will support for hashedrekord. This PR _only_ makes it possible to pre-compute _a_ hash supported by rekor. Right it only supports sha256 because rekor only supports that hash.

Let me know if I'm misunderstanding something. Thanks

Reply:
Please also see https://github.com/trailofbits/sigstore-rekor-types/issues/22